datafusion/physical-plan/src/filter.rs

```diff
@@ -729,11 +777,11 @@ impl Stream for FilterExecStream {
                         }).and_then(|(array, batch)| {
                             match as_boolean_array(&array) {
                                 Ok(filter_array) => {
-                                    self.metrics.selectivity.add_part(filter_array.true_count());
                                     self.metrics.selectivity.add_total(batch.num_rows());
-
-                                    self.batch_coalescer.push_batch_with_filter(batch.clone(), filter_array)?;
-                                    Ok(())
+                                    // TODO: support push_batch_with_filter in LimitedBatchCoalescer
```

Comment:
this is a good todo -- it would be good to file as a follow on ticket / PR perhaps

I can do it if you like

Reply:
Here is a PR: https://github.com/apache/datafusion/pull/18783